packages/@aws-cdk/aws-codepipeline-actions/lib/s3/deploy-action.ts

```diff
@@ -29,6 +29,21 @@ export interface S3DeployActionProps extends codepipeline.CommonAwsActionProps {
    * The Amazon S3 bucket that is the deploy target.
    */
   readonly bucket: s3.IBucket;
+
+  /**
+   * The specified canned ACL to objects deployed to Amazon S3.
+   * This overwrites any existing ACL that was applied to the object.
+   *
+   * @default - the original object ACL
+   */
+  readonly accessControl?: s3.BucketAccessControl;
+
+  /**
+   * The caching behavior for requests/responses for objects in the bucket.
+   *
+   * @default - none, decided to the HTTP client
+   */
+  readonly cacheControl?: CacheControlOptions;
```

Comment:
One question - does `CacheControlOptions` represent 100% of the capabilities of RFC-2616 response header values? My original design had an option of an "escape hatch" that would allow you to specify the string directly. This design no longer has that option. So, my question is, is the entire surface area of the RFC covered here?

Reply:
The escape hatch is still present in the form of the `of` property:

https://github.com/aws/aws-cdk/blob/7bf6857beec55cef6c9a54c2ebe0609399727ade/packages/%40aws-cdk/aws-codepipeline-actions/lib/s3/deploy-action.ts#L149-L154

Reply:
OK, got it. Here's my take:

1. I don't like the name `of`. I would call it `literal`.
2. I don't see why it should be an array, that seems confusing to me. I would turn it to a string.
3. I don't like the fact that whatever you pass here is concatenated at the end of the literal. I see 2 ways out of this:
   a) add a runtime check that either only `literal` has been passed (without any other properties), or it has not.
   b) have something like `CacheControl.of({ maxAge: ... })` and `CacheControl.literal('my string here')`, separate.
  Feel free to choose whichever one you like more.